docs/ocean/design_docs/min_level_cell.rst

```diff
@@ -0,0 +1,450 @@
+
+  * [ ] ``ocn_time_integration_split_init()``
+
+  * [ ] ``ocn_time_integrator_split()``
+
+* [ ] ``ocn_mesh``:
+
+  * [ ] ``minLevelCell``
+
+  * [ ] ``minLevelEdgeTop``
+
+  * [ ] ``minLevelEdgeBot``
+
+  * [ ] ``minLevelVertexTop``
+
+  * [ ] ``minLevelVertexBot``
+
+  * [ ] ``ocn_meshCreate()``
+
+  * [ ] ``ocn_meshUpdateFields()``
+
+  * [ ] ``ocn_meshDestroy()``
+
+* [ ] ``ocn_diagnostics``:
+
+  * [ ] ``ocn_relativeVorticity_circulation()``
+
+  * [ ] ``ocn_diagnostic_solve_layerThicknessEdge()``
+
+  * [ ] ``ocn_diagnostic_solve_vorticity()``
+
+  * [ ] ``ocn_diagnostic_solve_richardson()``
+
+  * [ ] ``ocn_diagnostic_solve_surfaceLayer()``
+
+  * [ ] ``ocn_diagnostic_solve_vortVel()``
+
+  * [ ] ``ocn_diagnostic_solve_z_coordinates()``
+
+  * [ ] ``ocn_diagnostic_solve_pressure()``
+
+  * [ ] ``ocn_vert_transport_velocity_top()``
+
+  * [ ] ``ocn_fuperp()``
+
+  * [ ] ``ocn_compute_land_ice_flux_input_fields()``
+
+  * [ ] ``ocn_validate_state()``
+
+...
+
+.. note::
+
+  ``ocn_equation_of_state_jm`` currently doesn't include any reference to
+  ``maxLevel*`` but this is mildly concerning.  T and S are clipped to the
+  valid range before density is computed, meaning that contamination with
+  invalid values could go unnoticed.  Still, it does appear that
+  ``displacedDensity`` is currently only used starting at index ``k = 2``, so
+  no invalid values should be getting used.
+
+Here is a (by no means complete) checklist of routines not used by
+``sub_ice_shelf_2D`` test cases that are a lower priority to update:
+
+Init:
+
+* [ ] ``ocn_init_cell_markers``:
+
+  * [ ] ``ocn_mark_maxlevelcell()``
+
+* [ ] ``ocn_init_global_ocean``:
+
+  * [ ] ``ocn_init_setup_global_ocean()``
+
+  * [ ] ``ocn_init_setup_global_ocean_create_model_topo()``
+
+  * [ ] ``ocn_init_setup_global_ocean_deepen_critical_passages()``
+
+  * [ ] ``ocn_init_setup_global_ocean_interpolate_land_ice_topography()``
+
+  * [ ] ``ocn_init_setup_global_ocean_modify_temp_under_land_ice()``
+
+  * [ ] ``ocn_init_setup_global_ocean_cull_inland_seas()``
+
+  * [ ] ``ocn_init_setup_global_ocean_interpolate_tracers()``
+
+* [ ] ``ocn_init_isomip``:
+
+  * [ ] ``ocn_init_setup_isomip()``
+
+* [ ] ``ocn_init_isomip_plus``:
+
+  * [ ] ``ocn_init_setup_isomip_plus()``
+
+Forward:
+
+* [ ] ``ocn_diagnostics``:
+
+  * [ ] ``ocn_filter_btr_mode_tend_vel()``
+
+* [ ] ``ocn_gm``:
+
+  * [ ] ``ocn_GM_compute_Bolus_velocity()``
+
+...
+
+
+Implementation: Surface fields and fluxes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+The subroutines  ``ocn_thick_surface_flux_tend()`` and
+``ocn_tracer_surface_flux_tend()`` already distribute surface fluxes over
+multiple vertical levels in a loop, so surface fluxes will simply require the
+same treatment as any other loops.
+
+It will be a little trickier to make sure we perform proper indexing of all
+3D variables to get their "surface" values.  Here, the plan is to make sure
```

Comment:
> It will be a little trickier to make sure we perform proper indexing of all
3D variables to get their "surface" values.
I'm not quite sure what you mean here. Am I correct in thinking that the surface values for cell-centered vars are at `minLevelCell` and the edge-centered vars at `minLevelEdgeBot`? Can you say more about where the trickiness comes in?

Reply:
I'll clarify in the document.  The trickiness is in finding them.  It's easy to search for `maxLevel` and find relevant loops.  It's quite a bit harder to usefully search for `1` or `2`, particularly if it's not associated with a loop over `k`.  These are variables where you're just indexing them at `1` without it being in a loop, so it would be easy to miss this and leave it as `1` instead of `minLevelCell(iCell)` or equivalent.